//
// PROJECT:         Aspia
// FILE:            protocol/authorization.proto
// LICENSE:         GNU General Public License 3
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto.auth;

enum Method
{
    METHOD_UNKNOWN = 0;
    METHOD_BASIC   = 1;
}

enum SessionType
{
    SESSION_TYPE_UNKNOWN        = 0;
    SESSION_TYPE_DESKTOP_MANAGE = 1;
    SESSION_TYPE_DESKTOP_VIEW   = 2;
    SESSION_TYPE_FILE_TRANSFER  = 4;
    SESSION_TYPE_SYSTEM_INFO    = 8;
}

enum Status
{
    STATUS_UNKNOWN       = 0;
    STATUS_SUCCESS       = 1;
    STATUS_ACCESS_DENIED = 2;
    STATUS_CANCELED      = 3;
}

message LogonRequest
{
    Method method = 1;
}

message ServerChallenge
{
    bytes nonce = 1;
}

message ClientChallenge
{
    SessionType session_type = 1;
    string username          = 2;
    bytes session_key        = 3;
}

message LogonResult
{
    Status status = 1;
}

message ClientToHost
{
    LogonRequest logon_request       = 1;
    ClientChallenge client_challenge = 2;
}

message HostToClient
{
    ServerChallenge server_challenge = 1;
    LogonResult logon_result         = 2;
}
